crates/kubelet/src/node.rs

```diff
@@ -444,16 +387,233 @@ fn node_labels_definition(arch: &str, config: &Config) -> HashMap<String, String
                 key
             );
         } else {
-            labels.insert(key.to_owned(), value.to_owned());
+            builder.add_label(key, value);
+        }
+    }
+}
+
+/// Kubernetes Node Definition. Wraps `k8s_openapi::api::core::v1::Node`.
+pub struct Node(k8s_openapi::api::core::v1::Node);
+
+impl Node {
+    /// Create builder for node definition.
+    pub fn builder() -> NodeBuilder {
+        Default::default()
+    }
+
+    /// Create node definition from `k8s_openapi::api::core::v1::Node` object.
+    pub fn from_k8s_openapi(node: k8s_openapi::api::core::v1::Node) -> Node {
```

Comment:
This should probably be done with `impl From<KubeNode> for Node`

Reply:
Agreed.